### PR TITLE
[Main] Allow for zPIV maintenance mode.

### PR DESCRIPTION
**Pull request by rejectedpromise**
Activate maintenance via SPORK_16.

Removed SPORK_17. Replaced with checks against ChainParams::Zerocoin_StartTime or ChainParams::Zerocoin_StartHeight

Review changes:

Disable pushButtonMintzPIV and pushButtonSpendzPIV by SPORK_16.

Added check for SPORK_16 in rpcwallet

add tooltip to disabled buttons

nTime to GetBlockTime()

### CODE REVIEW

File: src/main.cpp [hunk 1]
```diff
@@ -1271,8 +1271,8 @@ std::list<libzerocoin::CoinDenomination> ZerocoinSpendListFromBlock(const CBlock
 
 bool CheckZerocoinMint(const uint256& txHash, const CTxOut& txout, CValidationState& state, bool fCheckOnly)
 {
-    if(!fCheckOnly && GetAdjustedTime() < GetSporkValue(SPORK_17_ENABLE_ZEROCOIN))
-        return state.DoS(100, error("CheckZerocoinMint(): Zerocoin transactions are not allowed yet"));
+    if(!fCheckOnly && chainActive.Height() >= Params().Zerocoin_StartHeight())
```
Comment (presstab) [inline src/main.cpp]: You changed the operator here to reject all zerocoin transactions that have ever occurred.

File: src/main.cpp [hunk 2]
```diff
@@ -1560,7 +1555,11 @@ bool AcceptToMemoryPool(CTxMemPool& pool, CValidationState& state, const CTransa
     if (pfMissingInputs)
         *pfMissingInputs = false;
 
-    if (!CheckTransaction(tx, GetAdjustedTime() > GetSporkValue(SPORK_17_ENABLE_ZEROCOIN), state))
+    //Temporarily disable zerocoin for maintenance
+    if (GetAdjustedTime() > GetSporkValue(SPORK_16_ZEROCOIN_MAINTENANCE_MODE) && tx.ContainsZerocoins())
+        return state.DoS(100, error("AcceptToMemoryPool : Zerocoin transactions are temporarily disabled for maintenance"));
```
Comment (presstab) [inline src/main.cpp]: Should also include reject code.
`return state.DoS(10, error("AcceptToMemoryPool : Zerocoin transactions are temporarily disabled for maintenance"), REJECT_INVALID, "bad-tx");`

File: src/main.cpp [hunk 3]
```diff
@@ -1780,7 +1779,8 @@ bool AcceptableInputs(CTxMemPool& pool, CValidationState& state, const CTransact
     if (pfMissingInputs)
         *pfMissingInputs = false;
 
-    if (!CheckTransaction(tx, GetAdjustedTime() > GetSporkValue(SPORK_17_ENABLE_ZEROCOIN), state))
+
+    if (!CheckTransaction(tx, chainActive.Height() < Params().Zerocoin_StartHeight(), state))
```
Comment (presstab) [inline src/main.cpp]: What the hell another switched operator

**Comment by Fuzzbawls** [comment]
Closing in favor of #388 that addresses feedback here.

**Pull request closed without merging**